pandas/core/series.py

```diff
@@ -901,10 +901,11 @@ def _get_values_tuple(self, key):
     def _get_values(self, indexer):
         try:
             return self._constructor(self._mgr.get_slice(indexer)).__finalize__(self)
-        except ValueError:
+        except (ValueError, AssertionError):
```

Comment:
The reason that the AssertionError is raised internally is in the end caused by DatetimeTZBlock allowing 2D values, while because being an ExtensionBlock it disables `ndim` validation.

We could actually fix the underlying reason, but that might mean that we need to start validating the `values.ndim`. Will check in a follow-up commit if that is feasable.

Reply:
See second commit that overrides `_check_ndim` for DatetimeTZBlock. This might be a more "correct" fix (I didn't run all tests yet, though), but not sure that is worth it ..

cc @jbrockmendel